R/fit.R

```diff
@@ -273,6 +273,36 @@ CmdStanMCMC <- R6::R6Class(
       } else {
         cat(paste(self$runset$procs$chain_output(id), collapse="\n"))
       }
+    },
+
+    draws = function(inc_warmup = FALSE) {
+      if (!length(self$output_files())) {
+        stop("No chains finished successfully. Unable to retrieve the fit.",
+             call. = FALSE)
+      }
+      if (is.null(private$draws_)) {
+        private$read_csv_()
+      }
+      if(inc_warmup && private$sampling_info_$save_warmup) {
+        posterior::bind_draws(private$warmup_draws_, private$draws_, along="iteration")
+      } else {
+        private$draws_
+      }
+    },
+
+    sampler_diagnostics = function(inc_warmup = FALSE) {
+      if (!length(self$output_files())) {
+        stop("No chains finished successfully. Unable to retrieve the fit.",
```

Comment:
This error will already be caught by `private$read_csv_()` so we can delete these lines.

Reply:
Removed.